test/new-e2e/tests/agent-platform/persisting-integrations/persisting_integrations_test.go

```diff
@@ -0,0 +1,212 @@
+	osVersion       = flag.String("osversion", "", "os version to test")
+	platform        = flag.String("platform", "", "platform to test")
+	architecture    = flag.String("arch", "", "architecture to test (x86_64, arm64))")
+	flavorName      = flag.String("flavor", "datadog-agent", "package flavor to install")
+	srcAgentVersion = flag.String("src-agent-version", "7", "start agent version")
+)
+
+var filesToMock = map[string]string{
+	"/opt/datadog-agent/.diff_python_installed_packages.txt":     diffPythonInstalledPackages,
+	"/opt/datadog-agent/.postinst_python_installed_packages.txt": postinstPythonInstalledPackages,
+	"/opt/datadog-agent/requirements-agent-release.txt":          requirementsAgentRelease,
+}
+
+type persistingIntegrationsSuite struct {
+	e2e.BaseSuite[environments.Host]
+	srcVersion string
+}
+
+func TestPersistingIntegrations(t *testing.T) {
+	platformJSON := map[string]map[string]map[string]string{}
+
+	err := json.Unmarshal(platforms.Content, &platformJSON)
+	require.NoErrorf(t, err, "failed to umarshall platform file: %v", err)
+
+	osVersions := strings.Split(*osVersion, ",")
+	t.Log("Parsed platform json file: ", platformJSON)
+
+	vmOpts := []ec2.VMOption{}
+	if instanceType, ok := os.LookupEnv("E2E_OVERRIDE_INSTANCE_TYPE"); ok {
+		vmOpts = append(vmOpts, ec2.WithInstanceType(instanceType))
+	}
+
+	for _, osVers := range osVersions {
+		osVers := osVers
+		if platformJSON[*platform][*architecture][osVers] == "" {
+			// Fail if the image is not defined instead of silently running with default Ubuntu AMI
+			t.Fatalf("No image found for %s %s %s", *platform, *architecture, osVers)
+		}
+
+		t.Run(fmt.Sprintf("test upgrade persisting integrations on %s %s", osVers, *architecture), func(tt *testing.T) {
+			flake.Mark(tt)
+			tt.Parallel()
+			tt.Logf("Testing %s", osVers)
+
+			osDesc := platforms.BuildOSDescriptor(*platform, *architecture, osVers)
+			vmOpts = append(vmOpts, ec2.WithAMI(platformJSON[*platform][*architecture][osVers], osDesc, osDesc.Architecture))
+
+			e2e.Run(tt,
+				&persistingIntegrationsSuite{srcVersion: *srcAgentVersion},
+				e2e.WithProvisioner(awshost.ProvisionerNoAgentNoFakeIntake(
+					awshost.WithEC2InstanceOptions(vmOpts...),
+				)),
+				e2e.WithStackName(fmt.Sprintf("upgrade-persisting-integrations-%s-test-%s-%v-%s", *srcAgentVersion, *flavorName, osVers, *architecture)),
+			)
+		})
+	}
+}
+
+func (is *persistingIntegrationsSuite) TestIntegrationPersistsWithFileFlag() {
+	VMclient := is.SetupTestClient()
+
+	startAgentVersion := is.SetupAgentStartVersion(VMclient)
+	is.InstallNVMLIntegration(VMclient)
+	is.PrepareMockedFiles(VMclient)
+
+	// set the flag to install third party deps
+	is.EnableInstallThirdPartyDepsFlag(VMclient)
+
+	upgradedAgentVersion := is.UpgradeAgentVersion(VMclient)
+	is.Require().NotEqual(startAgentVersion, upgradedAgentVersion)
+	is.CheckIntegrationInstalled(VMclient)
+}
+
+// func (is *persistingIntegrationsSuite) TestIntegrationDoesNotPersistWithoutFileFlag() {
```

Comment:
We should test this use case

Reply:
I commented it out because it requires cleanup, both tests are run on the same environment.